iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalGrammar.java

```diff
@@ -128,74 +133,66 @@ public static LexerlessGrammarBuilder createGrammarBuilder() {
 
   private static void punctuators(LexerlessGrammarBuilder b) {
     for (Punctuator p : Punctuator.values()) {
-      b.rule(p).is(SPACING, p.getValue()).skip();
+      b.rule(p).is(p.getValue()).skip();
     }
   }
 
   private static void lexical(LexerlessGrammarBuilder b) {
-
-    b.rule(WHITESPACE_OR_ESCAPED_LINE_BREAK).is(
-      b.skippedTrivia(b.regexp("(?:[" + LexicalConstant.WHITESPACE + "]|" + DockerLexicalConstant.LINE_BREAK + ")*+"))
-    );
-
-    b.rule(WHITESPACE_OR_LINE_BREAK).is(
-      b.skippedTrivia(b.regexp("[" + LexicalConstant.LINE_TERMINATOR + LexicalConstant.WHITESPACE + "]*+"))
-    );
-
+    b.rule(WHITESPACE).is(b.regexp("["+LexicalConstant.WHITESPACE+"]++"));
+    b.rule(WHITESPACE_OR_LINE_BREAK).is(b.skippedTrivia(b.regexp("["+LexicalConstant.WHITESPACE+LexicalConstant.LINE_TERMINATOR+"]++")));
+    b.rule(EOL).is(b.regexp("(?:"+DockerLexicalConstant.EOL+"|$)"));
     b.rule(SPACING).is(
-      WHITESPACE_OR_ESCAPED_LINE_BREAK,
-      b.zeroOrMore(
-        b.commentTrivia(b.regexp(DockerLexicalConstant.COMMENT)), b.regexp(DockerLexicalConstant.EOL),
-        WHITESPACE_OR_ESCAPED_LINE_BREAK)
-    ).skip();
-
-    b.rule(INSTRUCTION_PREFIX).is(
-      WHITESPACE_OR_LINE_BREAK,
-      b.zeroOrMore(
-        b.commentTrivia(b.regexp(DockerLexicalConstant.COMMENT)),
-        WHITESPACE_OR_LINE_BREAK)
+      b.oneOrMore(
+        b.firstOf(
+          b.commentTrivia(b.regexp(DockerLexicalConstant.COMMENT)),
+          WHITESPACE_OR_LINE_BREAK
+        )
+      )
     ).skip();
 
     b.rule(EOF).is(b.token(GenericTokenType.EOF, b.endOfInput())).skip();
 
-    b.rule(STRING_LITERAL).is(SPACING, b.regexp(DockerLexicalConstant.STRING_LITERAL));
-    b.rule(STRING_UNTIL_EOL).is(SPACING, b.regexp(DockerLexicalConstant.STRING_UNTIL_EOL));
-    b.rule(STRING_LITERAL_WITH_QUOTES).is(SPACING, b.regexp(DockerLexicalConstant.STRING_LITERAL_WITH_QUOTES));
+    // TODO : those elements will be removed in the next grammar progressively, implicit spacing has been removed until then
+    b.rule(STRING_LITERAL).is(b.regexp(DockerLexicalConstant.STRING_LITERAL));
+    b.rule(STRING_UNTIL_EOL).is(b.regexp(DockerLexicalConstant.STRING_UNTIL_EOL));
+    b.rule(STRING_LITERAL_WITH_QUOTES).is(b.regexp(DockerLexicalConstant.STRING_LITERAL_WITH_QUOTES));
 
     b.rule(EQUALS_OPERATOR).is(b.regexp(DockerLexicalConstant.EQUALS_OPERATOR));
 
-    b.rule(KEY_IN_KEY_VALUE_PAIR_IN_EQUALS_SYNTAX).is(SPACING, b.regexp(DockerLexicalConstant.KEY_IN_KEY_VALUE_PAIR_IN_EQUALS_SYNTAX));
+    b.rule(KEY_IN_KEY_VALUE_PAIR_IN_EQUALS_SYNTAX).is(b.regexp(DockerLexicalConstant.KEY_IN_KEY_VALUE_PAIR_IN_EQUALS_SYNTAX));
```

Comment:
Why do you remove this from here? You can simply replace it with `WHITESPACE`. So you don't have to change every TreeFactory method?